cfgov/login/tests/test_auth.py

```diff
@@ -0,0 +1,213 @@
+from unittest.mock import patch
+
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Group, Permission
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.messages import get_messages
+from django.contrib.messages.middleware import MessageMiddleware
+from django.contrib.sessions.middleware import SessionMiddleware
+from django.test import TestCase, override_settings
+from django.test.client import RequestFactory
+
+from login.auth import (
+    CFPBOIDCAuthenticationBackend,
+    ensure_wagtail_users_group_membership,
+    process_family_name_claim,
+    process_given_name_claim,
+    process_roles_admin,
+    process_roles_claim,
+    username_from_email,
+)
+
+
+User = get_user_model()
+
+
+class UsernameFromEmailTestCase(TestCase):
+
+    def test_email_address(self):
+        self.assertEqual(username_from_email("test@example.com"), "test")
+
+
+class ProcessRolesClaimTestCase(TestCase):
+
+    def setUp(self):
+        self.user = User.objects.create(
+            username="user1", email="email@example.com", first_name="User"
+        )
+        self.group = Group.objects.get_or_create(name="Wagtail Users")[0]
```

Comment:
Because this is created in a database migration, this should always exist.

```suggestion
        self.group = Group.objects.get(name="Wagtail Users")
```

The only thing is, if we do this, we also need to make this change in order to be able to run tests without migrations using `SKIP_DJANGO_MIGRATIONS=1` (which is my preferred method):

```diff
diff --git a/cfgov/core/testutils/runners.py b/cfgov/core/testutils/runners.py
index 2383dd911..d3382f2a2 100644
--- a/cfgov/core/testutils/runners.py
+++ b/cfgov/core/testutils/runners.py
@@ -39,7 +39,7 @@ class TestRunner(DiscoverSlowestTestsRunner):
         # happen if e.g. we're only running tests that don't require one.
         if dbs:
             # Some required Wagtail data (like the default site) are created in
-            # Wagtail migrations. If we're skipping migrations when running
+            # data migrations. If we're skipping migrations when running
             # tests, we need to create this data ourselves.
             if settings.SKIP_DJANGO_MIGRATIONS:
                 self.initial_wagtail_data()
@@ -52,11 +52,12 @@ class TestRunner(DiscoverSlowestTestsRunner):
     def initial_wagtail_data(self):
         """Populate required Wagtail data.
 
-        This logic comes from these Wagtail migrations:
+        This logic comes from these migrations:
 
         - wagtailcore.0001_squashed_0016_change_page_url_path_to_text_field
         - wagtailcore.0025_collection_initial_data
         - wagtailcore.0054_initial_locale
+        - login.0005_add_default_user_group
         """
         # Create default Locale object.
         Locale.objects.create(
@@ -110,6 +111,7 @@ class TestRunner(DiscoverSlowestTestsRunner):
         # Create default groups.
         Group.objects.get_or_create(name="Editors")
         Group.objects.get_or_create(name="Moderators")
+        Group.objects.get_or_create(name="Wagtail Users")
 
 
 class StdoutCapturingTestRunner(TestRunner):
diff --git a/cfgov/login/tests/test_auth.py b/cfgov/login/tests/test_auth.py
index b34051344..dc52b8b9c 100644
--- a/cfgov/login/tests/test_auth.py
+++ b/cfgov/login/tests/test_auth.py
@@ -35,7 +35,7 @@ class ProcessRolesClaimTestCase(TestCase):
         self.user = User.objects.create(
             username="user1", email="email@example.com", first_name="User"
         )
-        self.group = Group.objects.get_or_create(name="Wagtail Users")[0]
+        self.group = Group.objects.get(name="Wagtail Users")
         self.admin_permission = Permission.objects.get(
             codename="access_admin",
             content_type=ContentType.objects.get(
```

What do you think?